_posts/2024-09-16-introducing-opensearch-2-17.md

```diff
@@ -0,0 +1,109 @@
+
+**Optimize query performance with improved multi-term query planning**
+
+In OpenSearch 2.12, we introduced a query planning feature for multi-term queries over keyword fields. These include `range` , `terms` , `wildcard` , `prefix`, and `fuzzy` queries. These queries tend be more expensive to evaluate on an index, because we need to collect all matching keyword terms for the field, then collect all documents that match all of the terms. With the change introduced in version 2.12, if one of these queries is part of a conjunction involving another clause that matches a few documents, we can avoid the heavy upfront work by applying the multi-term query as a secondary filter to documents that match the other clause. Unfortunately, [we have seen some cases](https://github.com/opensearch-project/OpenSearch/issues/14755) where this query planning has chosen the wrong path due to faulty cost estimation. While we work toward improving the cost estimates, we have reverted to the pre-2.12 behavior by default in version 2.17. You can enable the pre-2.12 behavior in versions 2.12--2.16 by setting `search.keyword_index_or_doc_values_enabled` to `true`.
+
+**Accelerate search performance with new techniques for computationally intensive queries**
+
+As part of our ongoing effort to improve search performance, we are introducing a new set of experimental features to improve performance of computationally expensive queries. Our approximation framework brings new techniques to short-circuit long-running queries by only scoring relevant documents in the query. The first queries to take advantage of this framework in version 2.17 are top-level range queries with no other clauses. These queries show benchmark results of up to 50x faster query running times with this optimization. To enable these new experimental features, set `opensearch.experimental.feature.approximate_point_range_query.enabled` to `true` in `opensearch.yml`.
+
+
+## Observability and log analytics
+
+This release provides an expansion of OpenSearch’s observability and log analytics capabilities.
+
+**Enhance trace analysis with a new custom trace source for advanced filtering and cross-cluster support**
```

Comment:
"analysis" => "analytics"? (not wrong as written, but "analytics" is used more frequently.

Reply:
Analysis meaning the process (i.e., analyzing)